src/test/java/org/prebid/server/bidder/impactify/ImpactifyBidderTest.java

```diff
@@ -0,0 +1,373 @@
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    private ImpactifyBidder impactifyBidder;
+
+    @Mock
+    private CurrencyConversionService currencyConversionService;
+
+    @Before
+    public void setUp() {
+        impactifyBidder = new ImpactifyBidder(TEST_ENDPOINT, jacksonMapper, currencyConversionService);
+    }
+
+    @Test
+    public void createBidderWithWrongEndpointShouldThrowException() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new ImpactifyBidder(INCORRECT_TEST_ENDPOINT,
+                jacksonMapper, currencyConversionService));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldConvertCurrencyIfNotDefault() {
+        // given
+        given(currencyConversionService.convertCurrency(any(), any(), anyString(), anyString()))
+                .willReturn(BigDecimal.TEN);
+
+        final BidRequest bidRequest = givenBidRequest(
+                impCustomizer -> impCustomizer.bidfloor(BigDecimal.ONE).bidfloorcur("EUR"));
+
+        //when
+        Result<List<HttpRequest<BidRequest>>> result = impactifyBidder.makeHttpRequests(bidRequest);
+
+        //then
+        assertThat(result.getErrors()).hasSize(0);
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+                .flatExtracting(BidRequest::getImp)
+                .extracting(Imp::getBidfloor, Imp::getBidfloorcur)
+                .containsExactly(tuple(BigDecimal.TEN, "USD"));
+    }
+
+    @Test
+    public void makeHttpRequestsWithValidDataWillThrowExceptionOnCurrencyConversion() {
+        // given
+        given(currencyConversionService.convertCurrency(any(), any(), anyString(), anyString()))
+                .willThrow(PreBidException.class);
+
+        final BidRequest bidRequest = givenBidRequest(
+                impCustomizer -> impCustomizer.bidfloor(BigDecimal.ONE).bidfloorcur("EUR"));
+
+        //when
+        Result<List<HttpRequest<BidRequest>>> result = impactifyBidder.makeHttpRequests(bidRequest);
+
+        //then
+        assertThat(result.getErrors()).hasSize(1)
+                .extracting(BidderError::getMessage)
+                .containsExactly("Unable to convert provided bid floor currency from EUR to USD for imp `123`");
+
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnValidBidResponseWithAllHeadersExceptIpv6() {
+        //given
+        final BidRequest bidRequest = givenBidRequest(
+                bidRequestCustomizer -> bidRequestCustomizer
+                        .device(givenDevice(deviceCustomizer -> deviceCustomizer.ua("ua").ip("ip").ipv6("ipv6")))
+                        .site(Site.builder().page("https://proper.web.site").build())
+                        .user(User.builder().buyeruid("buyer_user_uid").build()),
+                singletonList(impCustomizer -> impCustomizer
```

Comment:
Time to replace old facioned calls to singletonList with shiny new List.of()? ;)